### ISSUE TITLE
getFoos properties list not strictly checked

### ISSUE THREAD

**Issue by wolfsage** [cyrusimap/cyrus-imapd#1846]
If you call getMessages and pass in junk values to the properties array (thisfieldisfake, datex) the call succeeds and returns a list of objects with only the ids list filled in.

The properties argument should probably be strictly checked and an invalidArguments error should be returned (detailing which properties were invalid perhaps).

**Comment by rsto** [comment]
The spec isn't totally clear on this one, and the relevant sections to this seem to be:

http://jmap.io/spec-core.html#crud-methods

> properties: String[]|null If supplied, only the properties listed in the array are returned for each 
> Foo object. If null, all properties of the object are returned. The id of the object is always 
> returned, even if not explicitly requested.

and http://jmap.io/spec-core.html#vendor-specific-extensions

> Individual services [...] may take the form of [...] extra properties on existing data types (which
> may also appear in arguments to methods that take property names) [...] the server MUST   
> ONLY expose these extensions if the client explicitly opts in. Without opt-in, the server MUST 
> [...] reject anything that does not conform to it as specified.

My take: Without a standardised process for opting in to extensions in JMAP, the client asking for the "datex" and "thisisfake" properties is as close as it gets to opt-in. It's just that our server doesn't support these properties. But other server implementations could!

A client implementation that speaks to two or more JMAP server implementation might rather not want to tune its requests for each backend. That way, it's up to the client to decide if a missing, extension-property is an error to handle, or just treat it silently as unset. Always returning the id indicates the client, that this object exists.

TL;DR: I recommend to keep the current behaviour, but I'm open to change it if brings benefit for client implementations.

**Comment by rsto** [comment]
@wolfsage has there been a decision on this one?

**Comment by wolfsage** [comment]
Ah, apologies. Let's ask Neil on this one

**Comment by neilj** [comment]
Take this to the jmap@ietf.org mailing list! Once we get consensus there, I'll update the spec.

**Comment by rsto** [comment]
So the spec got heavily refactored since last time we looked at this, and the relevant section now seems to be definition of the core Foo/get method. Which doesn't define how to handle unknown properties.

@wolfsage Would you like to bring this to the mailing list? As it currently stands, I treat this as no-fix.

**Comment by wolfsage** [comment]
Sure, will take a look today

**Comment by wolfsage** [comment]
Discussion started here: https://mailarchive.ietf.org/arch/msg/jmap/GUL4hC2MtG_yM5QGZpC6YSPcIbE

**Comment by ksmurchison** [comment]
jmap-core-07 states the following:

If an invalid property is requested, the call MUST be rejected with an "invalidArguments" error.
